src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -13979,5 +13948,278 @@ int32[] arg2
+                // (4,14): error CS0029: Cannot implicitly convert type 'System.Func<int[], int>' to '<anonymous delegate>'
+                // withParams = noParams; // 2
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "noParams").WithArguments("System.Func<int[], int>", "<anonymous delegate>").WithLocation(4, 14));
+        }
+
+        [Fact]
+        public void ParamsArray_DelegateConversions_MethodGroups()
+        {
+            var source = """
+                int MethodNoParams(int[] xs) => xs.Length;
+                int MethodWithParams(params int[] xs) => xs.Length;
+                var noParams = MethodNoParams;
+                var withParams = MethodWithParams;
+                noParams = withParams; // 1
+                withParams = noParams; // 2
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (5,12): error CS0029: Cannot implicitly convert type '<anonymous delegate>' to 'System.Func<int[], int>'
+                // noParams = withParams; // 1
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "withParams").WithArguments("<anonymous delegate>", "System.Func<int[], int>").WithLocation(5, 12),
+                // (6,14): error CS0029: Cannot implicitly convert type 'System.Func<int[], int>' to '<anonymous delegate>'
+                // withParams = noParams; // 2
+                Diagnostic(ErrorCode.ERR_NoImplicitConv, "noParams").WithArguments("System.Func<int[], int>", "<anonymous delegate>").WithLocation(6, 14));
+        }
+
+        [Fact]
+        public void ParamsArray_LambdaConversions()
+        {
+            var source = """
+                int MethodNoParams(int[] xs) => xs.Length;
+                int MethodWithParams(params int[] xs) => xs.Length;
+                var noParams = MethodNoParams;
+                var withParams = MethodWithParams;
+                noParams = (params int[] xs) => xs.Length; // 1
+                noParams = (int[] xs) => xs.Length;
+                withParams = (params int[] xs) => xs.Length;
+                withParams = (int[] xs) => xs.Length;
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (5,26): warning CS9502: Parameter 1 has params modifier in lambda but not in target delegate type.
+                // noParams = (params int[] xs) => xs.Length; // 1
+                Diagnostic(ErrorCode.WRN_ParamsArrayInLambdaOnly, "xs").WithArguments("1").WithLocation(5, 26));
+        }
+
+        [Fact]
+        public void ParamsArray_MethodGroupConversions()
+        {
+            var source = """
+                int MethodNoParams(int[] xs) => xs.Length;
+                int MethodWithParams(params int[] xs) => xs.Length;
+                var lambdaNoParams = (params int[] xs) => xs.Length;
+                var lambdaWithParams = (int[] xs) => xs.Length;
```

Comment:
>lambdaWithParams [](http://example.com/codeflow?start=20&length=16)

It looks like the names are reversed - this lambda expression has no `params`.

Reply:
Good catch, fixed: bd9752c341ae39adac1a8d9334c595ecfc1b7bcb